src/WinSvc.cc

```diff
@@ -965,7 +965,7 @@ static int Win32SockInit(void)
         } else {
             opt = opt | SO_SYNCHRONOUS_NONALERT;
 
-            if (::setsockopt(INVALID_SOCKET, SOL_SOCKET, SO_OPENTYPE, (char *) &opt, optlen)) {
+            if (::setsockopt(INVALID_SOCKET, SOL_SOCKET, SO_OPENTYPE, reinterpret_cast<char *>(&opt), optlen)) {
```

Comment:
As demonstrated in PR #1373 this is a deceptive request. Acceding will result in a fall down the slippery slope of ever increasing perfection requests. Throwing this PR further out of the intended scope of fixing a **few** MinGW compile errors.

Reply:
> As demonstrated in PR #1373 this is a deceptive request. Acceding will result in a fall down the slippery slope of ever increasing perfection requests. Throwing this PR further out of the intended scope of fixing a **few** MinGW compile errors.

I disagree with the above assessment and speculation. However, may not need to agree on this to make progress: If you refactor this PR so that it is just "fixing a few MinGW compile errors", without modifying so many setsockopt() calls (most of which are not specific and some are probably not even applicable to MinGW builds), then this change request will become stale/inapplicable and can be resolved on those grounds. I do not know whether (and am not implying that) such refactoring is possible.

Reply:
AFAICT, [this change request](https://github.com/squid-cache/squid/pull/1359#discussion_r1204726369) has not been addressed when PR branch was force-pushed from 02b82a1 to 213df95 and my re-review requested a few hours ago.

Reply:
It was addressed by authors decision not to implement your requested change.

Reply:
> > AFAICT, [this change request](https://github.com/squid-cache/squid/pull/1359#discussion_r1204726369) has not been addressed when PR branch was force-pushed from [02b82a1](https://github.com/squid-cache/squid/commit/02b82a17493c4097e5ba316f6a44c3ed6d4706a1) to [213df95](https://github.com/squid-cache/squid/commit/213df95d46de47975799e6a4879eca5ac93245c5) and my re-review requested a few hours ago.

> It was addressed by authors decision not to implement your requested change.

Thank you for disclosing that decision; I was not aware of that decision when my re-review was requested.

This change request stands. Doing nothing does not address my blocking concern.